package com.nocta.myown.security;

import org.springframework.security.core.userdetails.User;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

import com.nocta.myown.entity.Usuario;
import com.nocta.myown.repository.UsuarioRepository;

@Service
public class AdminUserDetailsService implements UserDetailsService {

    private final UsuarioRepository usuarioRepository;

    public AdminUserDetailsService(UsuarioRepository usuarioRepository) {
        this.usuarioRepository = usuarioRepository;
    }

    @Override
    public UserDetails loadUserByUsername(String email)  throws UsernameNotFoundException {

        Usuario usuario = usuarioRepository.findByEmail(email)
                .orElseThrow(() ->
                        new UsernameNotFoundException( "Usuario no encontrado"
                        )
                );

        return User.withUsername(usuario.getEmail())
                .password(usuario.getPasswordHash())
                .roles(usuario.getRol().name())
                .disabled(Boolean.FALSE.equals(usuario.getActivo()))
                .build();
    }
}